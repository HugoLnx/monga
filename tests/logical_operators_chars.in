void orOperations() {
	char a, b, c;
	a = 10;
	b = 20;
	c = 30;
	if (a > c || c < b) {
		printf("a > c || c < b\n");
	} else {
		printf("! a > c || c < b\n");
	}

	if (c > a) {
		printf("c > a\n");
	} else {
		printf("! c > a\n");
	}

	if (!(a == a || b == b)) {
		printf("!(a == a || b == b)\n");
	} else {
		printf("a == a || b == b\n");
	}
	if (!(a != a || b == b)) {
		printf("!(a != a || b == b)\n");
	} else {
		printf("a != a || b == b\n");
	}
	if (!(a != a || b != b)) {
		printf("!(a != a || b != b)\n");
	} else {
		printf("a != a || b != b\n");
	}
}

void andOperations() {
	char a, b, c;
	a = 10;
	b = 20;
	c = 30;

	if (c > b && c >= c) {
		printf("c > b && c >= c\n");
	}
	if (c < b && c >= c) {
		printf("c > b && c >= c\n");
	} else {
		printf("! c < b && c >= c\n");
	}
	if ((c - a) > c && c >= c){
		printf("(c - a) > c && c >= c\n");
	} else {
		printf("! (c - a) > c && c >= c\n");
	}
	if (c > b && a <= c) {
		printf("c > b && a <= c\n");
	} else {
		printf("! c > b && a <= c\n");
	}
	if (!(a == a && b == b)) {
		printf("!(a == a && b == b)\n");
	} else {
		printf("a == a && b == b\n");
	}
	if (!(a != a && b == b)) {
		printf("!(a != a && b == b)\n");
	} else {
		printf("a != a && b == b\n");
	}
}

int main() {
	orOperations();
	andOperations();
}
