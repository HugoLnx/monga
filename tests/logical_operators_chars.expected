! a > c || c < b
c > a
c > b && c >= c
! c < b && c >= c
! (c - a) > c && c >= c
c > b && a <= c
